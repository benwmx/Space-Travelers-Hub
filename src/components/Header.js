import React from 'react';
import { NavLink } from 'react-router-dom';
import { useSelector } from 'react-redux';
import logo from '../img/logo.svg';
import Styles from './header.module.scss';

const Header = () => {
  const selectedRockets = useSelector((state) => state.rockets.filter((rocket) => rocket.reserved));
  const selectedMissions = useSelector(
    (state) => state.missions.filter((mission) => mission.joined),
  );
  return (
    <header className={Styles.header}>
      <div className={Styles.logo}>
        <img src={logo} alt="Yellow Planet" style={{ width: '50px', height: '50px' }} />
        <h1 className={Styles.title}>Space Travelers&apos; Hub</h1>
      </div>
      <div className={Styles.navLinks}>
        <NavLink to="/" activeClassName={Styles.activeLink} exact>
          Rockets
        </NavLink>
        <NavLink to="/missions" activeClassName={Styles.activeLink} exact>
          Missions
        </NavLink>
        <NavLink to="/myprofile" activeClassName={Styles.activeLink} exact>
          My profile
          <span className={Styles.numberOfSelected}>
            R(
            {selectedRockets.length}
            ) M (
            {selectedMissions.length}
            )
          </span>
        </NavLink>
      </div>
    </header>
  );
};

export default Header;
