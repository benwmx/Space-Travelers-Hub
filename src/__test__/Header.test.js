import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Header from '../components/Header';

describe('Header Component test', () => {
  it('Should render', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
