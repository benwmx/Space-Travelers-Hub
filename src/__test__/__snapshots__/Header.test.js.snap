// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component test Should render 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    <img
      alt="Yellow Planet"
      src="logo.svg"
      style={
        Object {
          "height": "50px",
          "width": "50px",
        }
      }
    />
    <h1
      className="title"
    >
      Space Travelers' Hub
    </h1>
  </div>
  <div
    className="navLinks"
  >
    <a
      aria-current="page"
      className="activeLink"
      href="#/"
      onClick={[Function]}
      style={Object {}}
    >
      Rockets
    </a>
    <a
      aria-current={null}
      href="#/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      aria-current={null}
      href="#/myprofile"
      onClick={[Function]}
    >
      My profile
      <span
        className="numberOfSelected"
      >
        R(
        0
        ) M (
        0
        )
      </span>
    </a>
  </div>
</header>
`;
